test/presenters/return-requirements/check/returns-requirements.presenter.test.js

```diff
@@ -0,0 +1,123 @@
+'use strict'
+
+// Test framework dependencies
+const Lab = require('@hapi/lab')
+const Code = require('@hapi/code')
+
+const { describe, it, beforeEach } = exports.lab = Lab.script()
+const { expect } = Code
+
+// Thing under test
+const ReturnRequirementsPresenter = require('../../../../app/presenters/return-requirements/check/returns-requirements.presenter.js')
+
+describe('Return Requirements presenter', () => {
+  let journey
+  let purposeIds = []
+  let requirement
+  let requirements = []
+
+  beforeEach(() => {
+    journey = {}
+
+    purposeIds = [{
+      id: '772136d1-9184-417b-90cd-91053287d1df',
+      description: 'A singular purpose'
+    }]
+
+    requirement = {
+      abstractionPeriod: {
+        'end-abstraction-period-day': '01',
+        'end-abstraction-period-month': '03',
+        'start-abstraction-period-day': '01',
+        'start-abstraction-period-month': '06'
+      },
+      agreementsExceptions: [
+        'gravity-fill'
+      ],
+      frequencyCollected: 'daily',
+      frequencyReported: 'daily',
+      points: [
+        '286'
+      ],
+      purposes: [
+        purposeIds[0].id
+      ],
+      returnsCycle: 'summer',
+      siteDescription: 'A place in the sun'
+    }
+
+    requirements = [{ ...requirement }]
+  })
+
+  describe('when provided requirements and purposes', () => {
+    it('correctly presents the data', () => {
+      const result = ReturnRequirementsPresenter.go(requirements, purposeIds, journey)
+
+      expect(result).to.equal({
+        returnsRequired: false,
+        requirements: [{
+          abstractionPeriod: 'From 1 June to 1 March',
+          frequencyCollected: 'daily',
+          frequencyReported: 'daily',
+          index: 0,
+          purposes: [
+            'A singular purpose'
+          ],
+          siteDescription: 'A place in the sun'
+        }]
+      })
+    })
+  })
+
+  describe("the 'requirements' property", () => {
+    describe('when the requirement has agreements exceptions', () => {
+      it('correctly returns and requirement with agreements exceptions', () => {
+        const result = ReturnRequirementsPresenter.go(requirements, purposeIds, journey)
+
+        expect(result.requirements).to.equal([{
+          abstractionPeriod: 'From 1 June to 1 March',
+          frequencyCollected: 'daily',
+          frequencyReported: 'daily',
+          index: 0,
+          purposes: [
+            'A singular purpose'
+          ],
+          siteDescription: 'A place in the sun'
+        }
+        ])
+      })
+    })
+    describe('when the requirement does not have any agreements exceptions', () => {
+      beforeEach(() => {
+        delete requirements[0].agreementsExceptions
+      })
+
+      it('does not return the requirement', () => {
+        const result = ReturnRequirementsPresenter.go(requirements, purposeIds, journey)
+
+        expect(result.requirements).to.equal([])
+      })
+    })
+    describe('has purpose id\'s', () => {
```

Comment:
NIT - whitespace

But, I'd also suggest moving this test into the `describe('when the requirement is "complete" (agreements exceptions is populated)')` block as we can assume a completed requirement will have purposes.

So, you'd end up with

```javascript
describe('when the requirement is "complete" (agreements exceptions is populated)', () => {
  it('correctly returns the requirement', () => {
  })

  it('maps the selected purpose ID\'s to their descriptions', () => {
  })
})
```

Reply:
Unless someone maliciously mucks about with the order that the journey is completed (in which case we blow up and on their head be it!) we will only call the functionality in the presenter to map purposes for a 'complete' requirement.

This means that code will _never_ be asked to format an empty array of purposes. So, I don't think there is any value in testing it. I've also double-checked the test output, and it shows no missing coverage for this presenter, so code coverage won't be a problem either.